.vscode/tasks.json

```diff
@@ -6,20 +6,98 @@
             "command": "dotnet",
             "type": "process",
             "args": [
-                "build",
+                "msbuild",
+                "-t:Build;Publish",
+                "${workspaceFolder}/src/minsk.sln"
+            ],
+            "problemMatcher": "$msCompile"
+        },
+        {
+            "label": "clean",
+            "command": "dotnet",
+            "type": "process",
+            "args": [
+                "clean",
                 "${workspaceFolder}/src/minsk.sln"
             ],
             "problemMatcher": "$msCompile"
         },
         {
-            "label": "mc",
+            "label": "build-msproj",
             "command": "dotnet",
-            "type": "shell",
+            "type": "process",
+            "options": {
+                "cwd": "${workspaceFolder}/samples",
+                "env": {
+                    "MinskToolsPath": "${workspaceFolder}/artifacts/Minsk",
+                }
+            },
+            "args": [
+                "build",
+                "${fileDirname}"
+            ],
+            "problemMatcher": "$msCompile"
+        },
+        {
+            "label": "run-msproj",
+            "command": "dotnet",
+            "type": "process",
+            "options": {
+                "cwd": "${workspaceFolder}/samples",
+                "env": {
+                    "MinskToolsPath": "${workspaceFolder}/artifacts/Minsk",
+                }
+            },
             "args": [
                 "run",
-                "--project",
-                "${workspaceFolder}/src/msc/msc.csproj",
-                "--",
+                "${fileDirname}"
+            ],
+            "problemMatcher": "$msCompile"
+        },
+        {
+            "label": "msc",
```

Comment:
This is currently broken due to references not being included, so it emits _type not resolved_ errors. One way to solve it, is by supporting `.rsp` response files to take defaults from it when building by msc alone. It will also be helpful for msi too.